src/Indicator.vala

```diff
@@ -27,14 +27,41 @@ public class Power.Indicator : Wingpanel.Indicator {
     private Services.Device primary_battery;
     private bool notify_battery = false;
 
+    private Widgets.ScreenBrightness screen_brightness;
```

Comment:
I am unsure why `screen_brightness` had to be moved here and you couldn't use a method on the `PopoverWidget` to propagate the value of the screen brigthness.

Reply:
Because I'm connecting `display_widget` with `on_icon_scroll_event`.

Reply:
That can be done with the code without moving.